server/base/src/main/java/org/apache/accumulo/server/conf/ServerConfigurationFactory.java

```diff
@@ -140,4 +158,114 @@ public void connectionEvent() {
+    }
+
+    /**
+     * Check that the stored version in ZooKeeper matches the version held in the local snapshot.
+     * When a mismatch is detected, a change event is sent to the prop store which will cause a
+     * re-load. If the Zookeeper node has been deleted, the local cache entries are removed.
+     * <p>
+     * This method is designed to be called as a scheduled task, so it does not propagate exceptions
+     * other than interrupted Exceptions so the scheduled tasks will continue to run.
+     */
+
+    private synchronized void verifySnapshotVersions() {
+      long refreshStart = System.nanoTime();
+      int keyCount = 0;
+      int keyChangedCount = 0;
+
+      PropStore propStore = context.getPropStore();
+      keyCount++;
+
+      // rely on store to propagate change event if different
+      propStore.validateDataVersion(SystemPropKey.of(context),
+          ((ZooBasedConfiguration) getSystemConfiguration()).getDataVersion());
+      // small yield - spread out ZooKeeper calls
+      jitterDelay();
+
+      for (Map.Entry<NamespaceId,NamespaceConfiguration> entry : namespaceConfigs.entrySet()) {
+        keyCount++;
+        PropStoreKey<?> propKey = NamespacePropKey.of(context, entry.getKey());
+        if (!propStore.validateDataVersion(propKey, entry.getValue().getDataVersion())) {
+          keyChangedCount++;
+          namespaceConfigs.remove(entry.getKey());
+        }
+        // small yield - spread out ZooKeeper calls between namespace config checks
+        jitterDelay();
+      }
+
+      for (Map.Entry<TableId,TableConfiguration> entry : tableConfigs.entrySet()) {
+        keyCount++;
+        TableId tid = entry.getKey();
+        PropStoreKey<?> propKey = TablePropKey.of(context, tid);
+        if (!propStore.validateDataVersion(propKey, entry.getValue().getDataVersion())) {
+          keyChangedCount++;
+          tableConfigs.remove(tid);
+          tableParentConfigs.remove(tid);
+          log.debug("data version sync: difference found. forcing configuration update for {}}",
+              propKey);
+        }
+        // small yield - spread out ZooKeeper calls between table config checks
+        jitterDelay();
+      }
+
+      log.debug("data version sync: Total runtime {} ms for {} entries, changes detected: {}",
+          NANOSECONDS.toMillis(System.nanoTime() - refreshStart), keyCount, keyChangedCount);
+    }
+
+    /**
+     * Generate a small random integer for jitter between [min,max).
+     *
+     * @param min
+     *          minimum number (inclusive)
+     *
+     * @param max
+     *          maximum number (exclusive)
+     *
+     * @return random number between min and MAJ_JITTER_DELAY
+     */
+    @SuppressFBWarnings(value = "PREDICTABLE_RANDOM",
+        justification = "random number not used in secure context")
+    private long jitter(final long min, final long max) {
+      return ThreadLocalRandom.current().nextLong(min, max);
+    }
```

Comment:
There's a lot of lines for the javadoc and boilerplate method, for a one-line method body that's only called in one place. I think it could be inline'd, and the javadoc isn't necessary, since the function is obvious once all the surrounding noise is removed.

Reply:
Partially incorporated in ac14691996 - I removed the extra documentation - this is used in two places, and trying to inline it caused spotbugs errors - it would not seem to recognize the suppress annotation unless it is on the method.